Libraries/LLM/Tokenizer.swift

```diff
@@ -51,7 +51,7 @@ private func updateTokenizerConfig(_ tokenizerConfig: Config) -> Config {
 }
 
 /// overrides for TokenizerModel/knownTokenizers
-let replacementTokenizers = [
+public var replacementTokenizers = [
```

Comment:
I just realized this isn't thread safe -- we get a warning with all the thread safety checks on and an error in swift 6.  We may need to wrap this up in an `actor`

Reply:
Oh my, there is indeed an error warning in Swift 6. If using actors, loadSynchronous may not be synchronous.https://github.com/ml-explore/mlx-swift-examples/blob/cf298551f317e33c390d035b94a3a9433abe9fcc/Libraries/LLM/Load.swift#L50

Reply:
Yes, that is the nature of mutable state -- you need a way to guard it.  We could potentially use a lock here as we don't expect much contention for it (indeed it is likely to be called once per run).  That would let us use unchecked Sendable and keep it synchronous.

Reply:
This has a nice writeup: https://stackoverflow.com/questions/76562442/in-swift-async-await-can-i-use-lock-or-semaphore

We have a nice tight critical section for these maps so something like NSLock should work well.